pkg/sessionctx/variable/session.go

```diff
@@ -1606,19 +1600,13 @@ func (s *SessionVars) IsPlanReplayerCaptureEnabled() bool {
 	return s.EnablePlanReplayerCapture || s.EnablePlanReplayedContinuesCapture
 }
 
-// GetNewChunkWithCapacity Attempt to request memory from the chunk pool
-// thread safety
-func (s *SessionVars) GetNewChunkWithCapacity(fields []*types.FieldType, capacity int, maxCachesize int, pool chunk.Allocator) *chunk.Chunk {
-	if pool == nil {
-		return chunk.New(fields, capacity, maxCachesize)
+// GetChunkAllocator returns a vaid chunk allocator.
+func (s *SessionVars) GetChunkAllocator() chunk.Allocator {
+	if s.chunkPool == nil {
+		return chunk.NewEmptyAllocator()
```

Comment:
Can we retun something like a `DefaultEmptyAllocator` instead?  (see also `context.Background()`)
Creating a `&emptyAllocator{}` is unnecessary and it increase object allocating